src/os/bluestore/BlueStore.cc

```diff
@@ -12441,6 +12441,12 @@ int BlueStore::set_collection_opts(
   if (c->pool_opts.get(pool_opts_t::COMPRESSION_REQUIRED_RATIO, &dval)) {
     c->compression_req_ratio = dval;
   }
+  int64_t csum_min_block;
+  if (c->pool_opts.get(pool_opts_t::CSUM_MIN_BLOCK, &csum_min_block)) {
```

Comment:
Wouldn't making a separate option be less confusing?
Besides here is another concern: generally the only reason why we don't remove this option is apparently a fear that this would break backward compatibility. Is that correct? If so then modifiing the internal logic behind the option makes things even worse - existing pools will silently stop writing data which results in data loss. IMO too bad...